### PR TITLE
GCS: Disable crash reporter when debugger connected

### LINKED ISSUE #1133: OSX: breakpad catches EXC_BREAKPOINT
This makes it incompatible with debugging generally, and causes it to crash the app immediately when starting a debug session through QtCreator.

### PR DESCRIPTION
**Pull request by tracernz**
Currently only on macOS (where it's a problem), and Windows (not sure?). Harder to do on Linux and not a problem there. 😛 

Fixes #1133

### CODE REVIEW

File: ground/gcs/src/libs/utils/hostosinfo.cpp [hunk 1]
```diff
@@ -56,3 +65,30 @@ HostOsInfo::HostArchitecture HostOsInfo::hostArchitecture()
...
+     * Based on Apple example "How do I determine if I'm being run under the debugger?"
+     * https://developer.apple.com/library/content/qa/qa1361/_index.html
+     */
+    int mib[4] = {CTL_KERN, KERN_PROC, KERN_PROC_PID, getpid()};
+    struct kinfo_proc info;
+    memset(&info, 0, sizeof(info));
+
+    // tell sysctl how much it has to play with
+    size_t size = sizeof(info);
+    if (sysctl(mib, sizeof(mib) / sizeof(*mib), &info, &size, NULL, 0) != 0) {
```
Comment (tracernz) [inline ground/gcs/src/libs/utils/hostosinfo.cpp:82]: I wonder if there's an array length macro in QtCore header.

**Comment by mlyle** [comment]
jenkins, build artifacts please

**Comment by tracernz** [comment]
build artifacts please

**Review by mlyle — APPROVED** [review]
LGTM-- just needs test.

**Comment by tracernz** [comment]
Just need to make sure it's good on Windows.

**Comment by tracernz** [comment]
Without debugger: http://dronin-autotown.appspot.com/at/crash/ahFzfmRyb25pbi1hdXRvdG93bnIWCxIJQ3Jhc2hEYXRhGICAgNiG_twLDA
Waiting for VS to install to test with debugger (entire base-devel group orders of magnitude faster to install on my arch boxes!!!).

**Comment by tracernz** [comment]
And MSVS doesn't even seem to be capable of debugging outside a project... wut... WinDbg to the rescue, and this PR is good (there was no crashreporter popup, and upon detaching WinDbg the standard Windows crash prompt came up). :)
drgcs-windbg

**Pull request merged**